### ISSUE TITLE
Compiler error, Data Type Wrong?

### ISSUE THREAD

**Issue by Zhu-jiatong** [MajenkoLibraries/SoftSPI#10]
I got the following compiler error when using the library in a project, seems to be an error with data type:
```
SoftSPI.cpp: In member function void SoftSPI::setClockDivider(uint8_t)
SoftSPI.cpp: 85:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV2
   ^~~~
SoftSPI.cpp: 88:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV4
   ^~~~
SoftSPI.cpp: 91:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV8
   ^~~~
SoftSPI.cpp: 94:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV16
   ^~~~
SoftSPI.cpp: 97:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV32
   ^~~~
SoftSPI.cpp: 100:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV64
   ^~~~
SoftSPI.cpp: 103:9: warning: case label value exceeds maximum value for type
   case SPI_CLOCK_DIV128
   ^~~~
SoftSPI.cpp: 86:20: warning: statement will never be executed [-Wswitch-unreachable]
   _delay = 2
   ~~~~~~~^~~
```

Additional info:
Board: ESP32 Dev Module

**Comment by Zhu-jiatong** [comment]
compiler ~error~ warning

**Comment by majenkotech** [comment]
That's a simple fix: the ESP32 uses some crazy-big numbers for those macros. I assume they're internal clock divider values that get directly applied to the SPI registers in normal operation. Simple expanding the parameter to a 32-bit integer is all that's needed.  Fixed in f069c19.
